scripts/zephyr_module.py

```diff
@@ -215,20 +220,52 @@ def main():
     cmake = ""
     sanitycheck = ""
 
+    dep_modules = []
+    sorted_modules = []
+    start_modules = []
+
     for project in projects:
         # Avoid including Zephyr base project as module.
         if project == os.environ.get('ZEPHYR_BASE'):
             continue
 
         meta = process_module(project)
         if meta:
-            kconfig += process_kconfig(project, meta)
-            cmake += process_cmake(project, meta)
-            sanitycheck += process_sanitycheck(project, meta)
+            section = meta.get('build', dict())
+            deps = section.get('depends', [])
+            if not deps:
+                start_modules.append( [project, meta] )
+            else:
+                dep_modules.append( [project, meta, deps] )
         elif project in extra_modules:
             sys.exit(f'{project}, given in ZEPHYR_EXTRA_MODULES, '
                      'is not a valid zephyr module')
 
+    # This will do a topology search to ensure the modules are ordered
```

Comment:
"topological" please

Reply:
and not search, should be sort.
I hope I do better coding than commenting when it gets late.

Reply:
fixed